### DISCUSSION TITLE
RelativeDistance in telemetry

### DISCUSSION THREAD

**Discussion by TracingInsights** [theOehrly/Fast-F1#490] [Ideas]
Currently 'RelativeDistance' is calculated as `rel_dist = d.loc[:, 'Distance'] / d.loc[:, 'Distance'].iloc[-1]` , I think it would be more accurate to use Circuit length as denominator like `rel_dist = d.loc[:, 'Distance'] / circuit_length` 

This has two advantages
1. We get to standardize it across different laps and drivers.a
2. we would be able to identify any potential data errors when the 'RelativeDistance' exceeds 1.

**Comment by theOehrly** [comment]
This was an intentional decision when implementing that method to compensate for slight integration errors.

Have you tested what happens if you do use a fixed distance instead? I'm sure, you will get values that are smaller and values that are larger than 1.0. But I wouldn't call these errors in the data but rather slight inaccuracies. (Except in very rare cases where the distance would be off by something like 10%, for example)

**Comment by TracingInsights** [comment]
As you pointed it out, I noticed 'RelativeDistance' highest value is almost always less than 1 when circuit length is used.

Regarding circuit length, Multiviewer mentioned circuit length data is available in F1TV API. Do you plan on adding circuit length data to Fast-F1 Api or Ergast Api? To save your time, I can compile a list and share it here if you'd like.

**Reply by theOehrly** [reply]
I'm not really sure if that is a reasonable approach. Why would we actually want to do that? The circuit lenght and the distance here are not the same really. If I remember correctly, the circuit length is measured along the centerline of the track. But the distance here is the distance on the racing line. 
I actually think that the current implementation is more useful than using a fixed circuit length for normalization.

**Reply by TracingInsights** [reply]
I think I didn't explain it correctly. The current approach is fine as is, I'm not suggesting any changes to 'RelativeDistance' metric.

In one of analysis, I calculate Average speed using driver's lap time and circuit length. For this I currently use 

` CIRCUIT_LENGTH = (
        f1session.laps.pick_fastest().get_car_data().add_distance().Distance.max()
    )`
    
   which works fine, but could be more accurate if I use the official circuit length. There are a few other instances as well where I needed circuit length for different calculations. It would be great if this data is available via FastF1

**Reply by theOehrly** [reply]
Ahh so what you want is access to the official circuit length and nothing else? Or am I still not understanding?

**Reply by TracingInsights** [reply]
Yes, that's it. I'll share the data for you and after you add it into Fast-F1 I'll create an example chart using circuit length in the docs

**Reply by theOehrly** [reply]
I don't really want to hard code this data into FastF1 if there is a better way. I'd prefer to use some API to access it. Else, this will just be one more thing that I need to remember to maintain manually. 
I'll see if I can't find this data somewhere else maybe.
